server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetVacationResponseMethod.java

```diff
@@ -42,6 +64,73 @@
 
     @Override
     public Stream<JmapResponse> process(JmapRequest request, ClientId clientId, MailboxSession mailboxSession) {
-        throw new NotImplementedException();
+        Preconditions.checkNotNull(request);
+        Preconditions.checkNotNull(clientId);
+        Preconditions.checkNotNull(mailboxSession);
+        Preconditions.checkArgument(request instanceof SetVacationRequest);
+        SetVacationRequest setVacationRequest = (SetVacationRequest) request;
+
+        if (!validateRequestStructure(setVacationRequest)) {
+            return Stream.of(JmapResponse
+                .builder()
+                .clientId(clientId)
+                .error(ErrorResponse.builder()
+                    .type(INVALID_ARGUMENTS1)
+                    .description(INVALID_ARGUMENT_DESCRIPTION)
+                    .build())
+                .build());
+        }
+
+        return process(clientId,
+            AccountId.create(mailboxSession.getUser().getUserName()),
+            setVacationRequest.getUpdate().get(Vacation.ID));
+    }
+
+    private boolean validateRequestStructure(SetVacationRequest setVacationRequest) {
+        return setVacationRequest.getUpdate().entrySet().size() == 1 && setVacationRequest.getUpdate().containsKey(Vacation.ID);
+    }
+
+    private Stream<JmapResponse> process(ClientId clientId, AccountId accountId, VacationResponse vacationResponse) {
+        boolean isValid = isValid(vacationResponse);
+
+        if (isValid) {
+            vacationRepository.modifyVacation(accountId, convertToVacation(vacationResponse));
+        }
+
+        return Stream.of(JmapResponse.builder()
+            .clientId(clientId)
+            .responseName(RESPONSE_NAME)
+            .response(generateSetVacationResponse(isValid, vacationResponse))
+            .build());
+    }
+
+    private boolean isValid(VacationResponse vacationResponse) {
+        return vacationResponse.getId().equals(Vacation.ID);
     }
+
+    public SetVacationResponse generateSetVacationResponse(boolean isValid, VacationResponse vacationResponses) {
+        if (isValid) {
+            return SetVacationResponse.builder()
+                .updatedId(Vacation.ID)
+                .build();
+        } else {
+            return SetVacationResponse.builder()
+                .notUpdated(ImmutableMap.of(Vacation.ID,
+                    SetError.builder()
+                        .type(INVALID_ARGUMENTS)
+                        .description(ERROR_MESSAGE_BASE + vacationResponses.getId())
+                        .build()))
+                .build();
+        }
+    }
+
+    public Vacation convertToVacation(VacationResponse vacationResponse) {
```

Comment:
A vacationResponse is the object defined in the JMAP spec. It then have a id which value is supposed to be set to "singleton"

Remark that the specs expects the user to use a mistaken id. This is the part of the spec supporting this : 

```
invalidArguments: Returned if one of the arguments is of the wrong type, or otherwise invalid (including using an id other than "singleton"). A description property MAY be present on the response object to help debug with an explanation of what the problem was.
```

The object I store in Cassandra had its id validated, and thus don't need to bother with this (saving place)

MoreOver using the same Vacation object for both JSON and storage would make me annotate a class of data-jmap with JSON annotations, which I strongly oppose. I prefer having duplication rather than JSON annotation on something that is not a model.

And if you suggest using the model to store data in Cassandra the problem is then circular dependency between data-jmap and jmap.